ai/core/src/unitycatalog/ai/core/oss.py

```diff
@@ -536,7 +547,188 @@ def delete_function(self, function_name: str, timeout: Optional[float] = None) -
 
     @override
     def to_dict(self) -> Dict[str, Any]:
-        return {k: getattr(self, k) for k in ["uc"] if getattr(self, k) is not None}
+        elements = ["functions_client", "catalogs_client", "schemas_client"]
+        return {k: getattr(self, k) for k in elements if getattr(self, k) is not None}
+
+    async def create_catalog_async(
```

Comment:
I feel like we should add these functions outside of this client. If we adopt https://github.com/unitycatalog/unitycatalog/pull/765/files#r1867316209 then we can add these functions in UnityCatalog class

Reply:
A most excellent idea! Will make this change :D